packages/bpk-component-checkbox/src/BpkCheckbox.js

```diff
@@ -70,12 +69,10 @@ const BpkCheckbox = props => {
         disabled={disabled}
         aria-label={label}
         aria-invalid={isInvalid}
+        // eslint-disable-next-line no-return-assign, no-param-reassign
+        ref={el => el && (el.indeterminate = indeterminate)}
```

Comment:
Using ref like this doesn't seem right. AFAIK refs are only called during "mount" and not update, so this will fail after the first render.

You should probably set the ref and use it in `componentDidUpdate` to set the indeterminate flag.

Reply:
Yeah there was not much online about setting this in react but from the react repo there was an issue opened to add support and both this and the approach mentioned above were acceptable until React adds support.

If the preferred way would be to do `componentDidUpdate` happy to address using that

Reply:
I believe either approach will come with side-effects until official support is added in react - according to the React engineers at FB

Reply:
Can you link to this issue? The problem here is that if this component is rendered with `indeterminate=false` and then the prop is updated after that, this will not work. So there is actually a bug, using the normal way of setting refs and updating this on `didUpdate` will ensure it works during "mount" and "update"

Reply:
`https://github.com/facebook/react/issues/1798`

Reply:
you can do something like this 
```
componentDidUpdate(props) {
  this.checkboxRef.indeterminate = props.indeterminate;
}
ref={el => {
   this.checkboxRef = el;
   this.checkboxRef.indeterminate = indeterminate;
}```

Reply:
Will we need `componentDidMount` as well?

Reply:
The `ref` will cover mount. But I tested this and it seems to work fine during update as well. 
We can keep it like it is, just remove the `eslint-disable` lines and follow the suggested style.
e.g:
```js
ref={e => {
  if (e) {
      e.indeterminate = indeterminate;
   }
}}

Reply:
Addressed